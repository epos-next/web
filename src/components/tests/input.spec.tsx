import React from "react"
import { fireEvent, render } from "@testing-library/react";
import Input from "@components/input";

it("should match snapshot", () => {
    const { asFragment } = render(<Input/>)
    expect(asFragment()).toMatchSnapshot()
});

describe("testing main logic", () => {
    it("should pass props to input-base", () => {
        const placeholder = "placeholder..."
        const myOwnProperty = "hello there"

        const { getByTestId } = render(<Input placeholder={ placeholder } data-myownproperty={ myOwnProperty }/>)

        expect(getByTestId("input-base")).toHaveAttribute("placeholder", placeholder)
        expect(getByTestId("input-base")).toHaveAttribute("data-myownproperty", myOwnProperty)
    });

    it("should use password icon if password=true", () => {
        const { asFragment } = render(<Input password={ true }/>)
        expect(asFragment()).toMatchSnapshot()
    });

    it("should show password if user click on eye icon", () => {
        const { getByTestId } = render(<Input password={ true }/>)
        fireEvent.click(getByTestId("password-eye"))
        expect(getByTestId("input-base")).toHaveAttribute("type", "text")
        expect(getByTestId("password-eye")).toHaveAttribute("src", "/icons/eye-icon_opened.png")
        fireEvent.click(getByTestId("password-eye"))
        expect(getByTestId("input-base")).toHaveAttribute("type", "password")
        expect(getByTestId("password-eye")).toHaveAttribute("src", "/icons/eye-icon_closed.png")
    });

    it("should render error text", () => {
        const error = "yooo this is my error::)"
        const { getByTestId } = render(<Input error={ true } errorText={ error }/>)
        expect(getByTestId("error")).toHaveTextContent(error)
    });

    it("shouldn't show error text if error=false and errorText != undefined", () => {
        const error = "yooo this is my error::)"
        const { getByTestId } = render(<Input error={ false } errorText={ error }/>)
        expect(getByTestId("input-base")).toHaveAttribute("data-error", "false")
    });
})

describe("testing props", () => {
    it("should use error", () => {
        const { getByTestId, rerender } = render(<Input error={ true }/>)
        expect(getByTestId("input-base")).toHaveAttribute("data-error", "true")
        rerender(<Input error={ false }/>)
        expect(getByTestId("input-base")).toHaveAttribute("data-error", "false")
    });

    it("should use errorText", () => {
        const { getByTestId, rerender } = render(<Input error={ true } errorText={ "text1" }/>)
        expect(getByTestId("error")).toHaveTextContent("text1")
        rerender(<Input error={ true } errorText={ "vchera ya govoril s bogom on menya ne ponyal" }/>)
        expect(getByTestId("error")).toHaveTextContent("vchera ya govoril s bogom on menya ne ponyal")
    });

    it("should use onEnterPressed", () => {
        const callback = jest.fn()
        const { getByTestId } = render(<Input onEnterPressed={ callback }/>)
        fireEvent.change(getByTestId("input-base"), { target: { value: "bump" } })
        fireEvent.keyDown(getByTestId("input-base"), { key: 'Enter', code: 'Enter' })
        expect(callback).toBeCalledTimes(1)
    });
})

it("should have input class", () => {
    const { getByTestId } = render(<Input/>)
    expect(getByTestId("container")).toHaveClass("input")
});
