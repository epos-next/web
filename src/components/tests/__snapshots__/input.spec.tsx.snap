// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c1 {
  width: 100%;
  border: 2px solid var(--light-primary);
  border-radius: 10px;
  outline: none;
  padding: 14px;
  font-size: 18px;
  color: var(--primary);
  -webkit-transition: 150ms border;
  transition: 150ms border;
}

.c1::-webkit-input-placeholder {
  color: var(--light-primary);
}

.c1::-moz-placeholder {
  color: var(--light-primary);
}

.c1:-ms-input-placeholder {
  color: var(--light-primary);
}

.c1::placeholder {
  color: var(--light-primary);
}

.c1:focus {
  border: 2px solid var(--contrast);
}

.c1[data-error=true] {
  color: var(--error);
  border: 2px solid var(--error);
}

.c1[data-error=true]:focus {
  border: 2px solid var(--error);
}

.c0 {
  width: 100%;
  position: relative;
}

@media screen and (max-width:768px) {
  .c1 {
    padding: 13px;
  }
}

<div
    class="c0 input "
    data-testid="container"
  >
    <input
      class="c1"
      data-error="false"
      data-testid="input-base"
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`testing main logic should use password icon if password=true 1`] = `
<DocumentFragment>
  .c2 {
  position: absolute;
  top: 16px;
  right: 14px;
  cursor: pointer;
}

.c4 {
  width: 100%;
  border: 2px solid var(--light-primary);
  border-radius: 10px;
  outline: none;
  padding: 14px;
  font-size: 18px;
  color: var(--primary);
  -webkit-transition: 150ms border;
  transition: 150ms border;
}

.c4::-webkit-input-placeholder {
  color: var(--light-primary);
}

.c4::-moz-placeholder {
  color: var(--light-primary);
}

.c4:-ms-input-placeholder {
  color: var(--light-primary);
}

.c4::placeholder {
  color: var(--light-primary);
}

.c4:focus {
  border: 2px solid var(--contrast);
}

.c4[data-error=true] {
  color: var(--error);
  border: 2px solid var(--error);
}

.c4[data-error=true]:focus {
  border: 2px solid var(--error);
}

.c0 {
  width: 100%;
  position: relative;
}

.c0 .c1[data-open=false] ~ .c3:not(:placeholder-shown) {
  -webkit-letter-spacing: 10px;
  -moz-letter-spacing: 10px;
  -ms-letter-spacing: 10px;
  letter-spacing: 10px;
  font-size: 14px;
  height: 52px;
}

@media screen and (max-width:768px) {
  .c4 {
    padding: 13px;
  }
}

<div
    class="c0 input "
    data-testid="container"
  >
    <img
      class="c1 c2"
      data-open="false"
      data-testid="password-eye"
      src="/icons/eye-icon_closed.png"
    />
    <input
      class="c3 c4"
      data-error="false"
      data-testid="input-base"
      type="password"
    />
  </div>
</DocumentFragment>
`;
